src/components/sections/DmEditSidebar.vue

```diff
@@ -1,8 +1,30 @@
 <template>
   <v-navigation-drawer class="px-2" floating :width="450" color="transparent">
     <v-card class="elevation-0">
+      <v-text-field class="pt-3 pl-3 mr-2" v-model="search" prepend-inner-icon="mdi-magnify" clearable label="Asignatura"
+        type="text" variant="outlined" :append-icon="'mdi-cog-outline'" @click:append="showFilter = !showFilter"
+        @click:clear="clearSearch">
+      </v-text-field>
```

Comment:
Play with paddings and margins could be overcomplicated. Maybe use a `v-container` to warp all the `v-text-field` and the advanced filter options